src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -87,12 +73,69 @@ public StaticHostProvider(Collection<InetSocketAddress> serverAddresses)
         Collections.shuffle(this.serverAddresses);
     }
 
+    /**
+     * In Java 7, we have a method getHostString, but earlier versions do not support it.
+     * This method is to provide a replacement for InetSocketAddress.getHostString().
+     *
+     * It evaluates to a hostname if one is available and otherwise it returns the
```

Comment:
nit: Can we swap the first two lines of this comment, as the reader does not know what class getHostString belongs to until he/she reaches line 2?